lib/active_ldap/connection.rb

```diff
@@ -136,7 +136,24 @@ def remove_connection(klass_or_key=self)
       end
 
       def setup_connection(config=nil)
+        if config.is_a?(Hash) and config.size == 1 and config.keys[0].to_sym == :name
```

Comment:
Do we need `.to_sym` here? Can we use `key?` instead? 

```suggestion
        if config.is_a?(Hash) and config.size == 1 and config.key?(:name)
```

Reply:
We need `.to_sym` here to support string keys, like 
```ruby
ActiveLdap::Base.setup_connection("name" => "special")
```

I tried it with `.key?`, but that fails

Reply:
Hmm, why do we need to support the case...?

The below `config[:name]` always uses `:name` as key. So if we use `config.keys[0].to_sym == :name` here, the below `config[:name]` will be failed.